cognite/neat/graph/stores/_base.py

```diff
@@ -355,8 +355,8 @@ def list_instances_of_type(self, class_uri: URIRef) -> list[ResultRow]:
             List of triples for instances of the given class
         """
         query = (
-            f"SELECT ?instance ?prop ?value "
-            f"WHERE {{ ?instance rdf:type <{class_uri}> . ?instance ?prop ?value . }} order by ?instance "
+            "SELECT ?instance ?prop ?value "
+            f"WHERE {{ ?instance <rdf:type> <{class_uri}> . ?instance ?prop ?value . }} order by ?instance "
```

Comment:
```suggestion
            f"WHERE {{ ?instance a <{class_uri}> . ?instance ?prop ?value . }} order by ?instance "
```

Reply:
Since RDF is not schema driven you might end up having a property for which you expect single value, but instead you have list, meaning you will have extra triples coming in, which probably will lead to breaking of pydantic classes.

Reply:
This will return you both Attributes of a node and Connections of a node, as in RDF properties are simply predicates of a triple. Alt is to explicitly via ontology which property is of type `DatatypeProperty` (attributes) and which `ObjectProperty` (connections). One can get along with `Filter` to select only properties which hold values:

```sparql
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>

SELECT ?property ?value
WHERE {
    ?subject ?property ?value .
    FILTER(isLiteral(?value))
}
```

or IRIs (connection):

```sparql
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>

SELECT ?property ?value
WHERE {
    ?subject ?property ?value .
    FILTER(isIRI(?value))
}
```

there are ways to construct simpler sparql query, merging lists, etc. , though as written above we are forced to transform source graph to singular namespace with ids of properties and classes that DMS compliant.

Somewhere we need to pay the price :)